Pod/Cache/GenericCoreDataStack.swift

```diff
@@ -185,14 +191,38 @@ public class GenericCoreDataStack<CoordinatorType: NSPersistentStoreCoordinator,
                 }
             }
             
-            logInfo(tag) {
-                "Attaching persistent store \"\(storeURL.lastPathComponent ?? "Unknown")\" for type: \(persistentStoreType)."
-            }
-            try persistentStoreCoordinator.addPersistentStoreWithType(storeType, configuration:  configuration, URL: storeURL, options: options)
-            
-            logInfo(tag) {
-                "Persistent store attached successfully."
+            logInfo(tag) { "Attaching persistent store \"\(storeURL.lastPathComponent ?? "Unknown")\" for type: \(persistentStoreType)."}
+
+            let persistentStore = try persistentStoreCoordinator.addPersistentStoreWithType(storeType, configuration:  configuration, URL: storeURL, options: options)
+
+            if options?[CCOverwriteIncompatibleModelOption] as? Bool == true {
+
+                let model = persistentStoreCoordinator.managedObjectModel
+                let metadata = persistentStoreCoordinator.metadataForPersistentStore(persistentStore)
+
+                logInfo(tag) { "Checking to see if persistent store is compatible with the model." }
+
+                if !model.isConfiguration(configuration, compatibleWithStoreMetadata: metadata) {
+
+                    logInfo(tag) { "Model is incompatible. Attempting to remove the persistent store." }
+                    try persistentStoreCoordinator.removePersistentStore(persistentStore)
+
+                    logInfo(tag) { "Attempting to remove file \(storeURL) and -shm and -wal files" }
+                    if let path = storeURL.path {
+
+                        let fileManager = NSFileManager.defaultManager()
+
+                        try fileManager.removeItemAtURL(storeURL)
+                        try fileManager.removeItemAtURL(NSURL(fileURLWithPath: "\(path)-shm"))
```

Comment:
@phchang We should handle the case where the file may not be present, suppose there was a -wal but not -shm.  If the removeItemAtURL is tried on the -shm, it jump to the catch, skipping the -wal.

Would a better approach be to get a listing of files that match the filename <filename>.* and delete what is found?